### DISCUSSION TITLE
How to enable Two sensor on single I2C port?

### DISCUSSION THREAD

**Discussion by ebhangar** [iNavFlight/inav#8992] [Q&A]
I have following hardware:
FC - Matek H743 Slim v3
GPS - Matek M10 5883
Radar - VL53L1X
Optical Flow - CXOF 

INAV version - 6.0

Problem: I'm facing problem with setting up compass. There are two I2C ports on this FC. One is already taken by Baro and other is available for user device. When I connect the compass to I2C with baro, INAV shows it red and when it is connected with Radar on second I2C port, both becomes red. However single sensor works fine on one I2C port.

Please help me out with how to enable/setup compass either with baro or with radar on single I2C bus/port.

**Comment by DzikuVx** [comment]
You do not have to enable anything. Most probably new compass is damaged or breaks I2C function in some way. Try different campass
